### ISSUE TITLE
Improve UX around .NET Core build asset generation (was: Unable to debug ASP.Net core)

### ISSUE THREAD

**Issue by mwgustin** [microsoft/vscode-docker#1760]
Docker extension version: 1.0.0

Steps to reproduce: 

1. Create a new project `dotnet new webapi` (dotnet 3.1)
2. Add docker files via extension command (Linux, Docker-compose yes, 80;443)
3. Try to debug with F5

Debug Console: 
```Starting: "docker" exec -i testproject-dev /remote_debugger/vsdbg --interpreter=vscode
-------------------------------------------------------------------
You may only use the Microsoft .NET Core Debugger (vsdbg) with
Visual Studio Code, Visual Studio or Visual Studio for Mac software
to help you develop and test your applications.
-------------------------------------------------------------------
  It was not possible to find any installed .NET Core SDKs
  Did you mean to run .NET Core SDK commands? Install a .NET Core SDK from:
      https://aka.ms/dotnet-download
The target process exited without raising a CoreCLR started event. Ensure that the target process is configured to use .NET Core. This may be expected if the target process did not run on .NET Core.
The program '[23] dotnet' has exited with code 145 (0x91).
```

My best guess is that this may have something to do with the docker-build: debug task targets the 'base' but the 'base' in the default Dockerfile is the production ASP image, not the 'build' SDK one.

Current environment details in case it matters, but I was able to reproduce on both Mac and Windows:
Version: 1.43.0
Commit: 78a4c91400152c0f27ba4d363eb56d2835f9903a
Date: 2020-03-09T19:34:44.548Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.2.0

**Comment by bwateratmsft** [comment]
Hey @mwgustin, is there any output in the Terminal window for the `dotnet build` / `docker build` / `docker run`?

**Comment by mwgustin** [comment]
Hey @bwateratmsft, that helped.  The terminal actually didn't show a `dotnet build` task at all...

...because, it looks like the issue is that if you do a new project, and don't initialize the project with VSCode (just add docker files via the extension), it will never create the default dotnet build task. The Docker extension relies on that default build task for debugging, but not for building/deploying within the docker container, so doing docker builds and docker-compose up works fine, only debugging fails. 

Not sure there's a good way around this since the docker extension probably doesn't want to be responsible for generating that default build task... maybe some checking/error logging if the user attempts to debug without that default build task existing? 

To get it working, I had to do `.NET: Generate assets for Build and Debug`, delete the docker files, then re-add the docker files because the .NET command overwrites the existing task.json and launch.json.

**Comment by bwateratmsft** [comment]
Aha, yeah that makes sense. Our documents do mention doing the `Generate assets for Build and Debug` step, unfortunately the C# extension completely obliterates any existing `tasks.json` and `launch.json` which necessitates re-running the `Add Docker Files...`

I'm going to keep this item open to consider what ways we can improve this user experience.

**Comment by bwateratmsft** [comment]
This is probably mitigated or fixed entirely by #1780

**Comment by bwateratmsft** [comment]
I'll call this fixed by #1780. We also have #1300 to track more improvements.
